modules/gateway/peers.go

```diff
@@ -35,6 +33,19 @@ var (
 			panic("unrecognized build.Release")
 		}
 	}()
+	// the gateway will abort a connection attempt after this long
+	dialTimeout = func() time.Duration {
+		switch build.Release {
+		case "dev":
+			return 2 * time.Minute
```

Comment:
dev is meant to be on your local machine, 10 seconds is definitely enough for a timeout.